File: java-kotlin-project/app/src/main/java/example/java/exception/ExceptionBasic.java
package example.java.exception;

import java.io.BufferedReader;
import java.io.FileNotFoundException;
import java.io.FileReader;
import java.io.IOException;

public class ExceptionBasic {

    public void divide(int a, int b) {
        try {
            int result = a / b;
            System.out.println("Result: " + result);
        } catch (ArithmeticException e) {
            System.out.println("Error: Cannot divide by zero.");
            e.printStackTrace();
        } finally {
            System.out.println("Division operation completed.");
        }
    }

    // 여러 예외 처리 (with Checked Exception)
    public void readFile(String path) {
        try {
            // 파일 읽기 로직
            FileReader reader = new FileReader(path);
            BufferedReader br = new BufferedReader(reader);
            String line = br.readLine();
            System.out.println(line);
        } catch (FileNotFoundException e) {
            System.out.println("File not found: " + path);
        } catch (IOException e) {
            System.out.println("Error reading file");
        } finally {
            // 리소스 정리
            System.out.println("Cleanup");
        }
    }

    // Java 7+ Multi-catch
    public void MultiCatch(String path) {
        try {
            FileReader reader = new FileReader(path);
            BufferedReader br = new BufferedReader(reader);
            String line = br.readLine();
            System.out.println(line);
        } catch (IOException | ArithmeticException e) {
            System.out.println("IO or Arithmetic error: " + e.getMessage());
        }
    }
}

File: java-kotlin-project/app/src/main/java/example/kotlin/exception/ExceptionBasic.kt
package example.kotlin.exception

import java.io.File
import java.io.FileNotFoundException
import java.io.IOException

class ExceptionBasic {

    // 기본 : kotlin에서 try-catch는 표현식이다.
    fun divide(a: Int, b: Int) {
        try {
            val result = a / b
            println("Result: $result")
        } catch (e: ArithmeticException) {
            println("Error: Cannot divide by zero.")
            e.printStackTrace()
        } finally {
            println("Division operation completed.")
        }
    }

    // try를 표현식으로 사용
    fun safeDivide(a: Int, b: Int): Int? {
        return try {
            a / b
        } catch (e: ArithmeticException) {
            null
        }
    }

    // 여러 예외 처리
    fun readFile(path: String) {
        try {
            val content = File(path).readText()
            println(content)
        } catch (e: FileNotFoundException) {
            println("File not found: $path")
        } catch (e: IOException) {
            println("Error reading file.")
        } finally {
            println("Completed.")
        }
    }

    // 표현식으로 기본값 제공 가능
    fun getFileContent(path: String): String {
        return try {
            File(path).readText()
        } catch (e: Exception) {
            "Default content..."
        }
    }
}